packages/agreement-process/test/agreementService.integration.test.ts

```diff
@@ -169,6 +172,42 @@ const expectedAgreementCreation = async (
   return actualAgreement;
 };
 
+const expectedAgreementArchived = async (
+  agreementId: AgreementId | undefined
+): Promise<AgreementV1> => {
+  expect(agreementId).toBeDefined();
+  if (!agreementId) {
+    fail("Unhandled error: returned agreementId is undefined");
+  }
+
+  const actualAgreementData: StoredEvent | undefined =
+    await readLastAgreementEvent(agreementId, postgresDB);
+
+  if (!actualAgreementData) {
+    fail("Creation fails: agreement not found in event-store");
+  }
+
+  expect(actualAgreementData).toMatchObject({
+    type: "AgreementUpdated",
+    event_version: 1,
+    version: "0",
+    stream_id: agreementId,
+  });
+
+  const actualAgreement: AgreementV1 | undefined = protobufDecoder(
+    AgreementUpdatedV1
+  ).parse(actualAgreementData.data)?.agreement;
+
+  if (!actualAgreement) {
+    fail("impossible to decode AgreementAddedV1 data");
+  }
+
+  expect(actualAgreement.state).toBe(
+    toAgreementStateV1(agreementState.archived)
+  );
```

Comment:
Could you test the entire expected agreement, not only the new state?
The archive should result in: 
- the state changing to archived
- the addition of an `archiving` stamp
- the rest of the agreement to be unchanged

So we should test that the agreement is the same but with these two changes